### ISSUE TITLE
waterBaseProp_pT contains undefined outputs

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#995]
**Reported by sjoelund&#46;se on 20 Feb 2013 12:05 UTC**
Modelica.Media.Water.IF97_Utilities.waterBaseProp_pT contains undefined outputs:
aux.dpT is never defined
For region 1, 2, 5: pt, pd, x (for 5)
For region 3: rho, vt, vp, cp

They should at the very least all be initialized to 0 to prevent usage of uninitialized variables.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/995

**Comment by modelica-trac-importer** [comment]
**Comment by sjoelund&#46;se on 20 Feb 2013 18:38 UTC**
Found one more warning for Media:

ModelicaTest.Media.TestAllProperties.LinearColdWater.err:
```
[Modelica 3.2.1/Media/package.mo:4815] Warning: state was used before it was defined (given a value). Additional such uses may exist for the variable, but some messages were suppressed.
```

If I read that warning correctly... state.T was used but is actually undefined since the statement is the one that initializes state? I could be confused by the inheritance though, so I asked omc and I agree with it that the code looks odd (state.T on rhs before state was defined):
```
function ModelicaTest.Media.TestAllProperties.LinearWater_pT_Ambient.Medium.setState_dTX
  ...
  output ModelicaTest.Media.TestAllProperties.LinearWater_pT_Ambient.Medium.ThermodynamicState state;
algorithm
  state := ModelicaTest.Media.TestAllProperties.LinearWater_pT_Ambient.Medium.ThermodynamicState(-2384310951.349984 + d / 0.0000004502096643198283 + 569451.032276449 * state.T, T);
end ModelicaTest.Media.TestAllProperties.LinearWater_pT_Ambient.Medium.setState_dTX;
```

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 6 Mar 2013 14:18 UTC**
Fixed in 835883d86a592cb54a594710a8e176ba8b8053a5.
